### DISCUSSION TITLE
Link Connection

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#2174] [Q&A]
### Introduction

![grafik](https://github.com/clientIO/joint/assets/46285156/c2a8d024-2434-4304-b5e3-a27ba547082d)

I have this custom dia.Element. It has the following markup:

```js
markup = [
  {
    tagName: "g",
    selector: "body",
    children: [
      {
        tagName: "line",
        selector: "line",
      },
      {
        tagName: "rect",
        selector: "header",
      },
    ],
  },
  {
    tagName: "text",
    selector: "label",
  },
];
```

My goal is for the dia.Links to connect either with the line or the header, depending on the connection position. Unfortunately, I can only manage to have the link connect with the header and the BBox (width of the header * length of the line). If I take the line as the body, then the link only connects with the line, of course. Does anyone have an idea how to solve this? :)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Before anything else, please create a JSFiddle with the above.

- You can set `magnet` attribute on both the `line` and `header`.
- You can represent the shape as a `dia.Element` and `dia.Link`.
- You can define the connection position via `anchor` (it can be located anywhere)
